### PR TITLE
[tensorflow] | [build, test] | [ec2] Add extra shm for TF Hvd GPU test on p2.8xlarge

### PR DESCRIPTION
**Pull request by YYStreet**
*Issue #, if available:*

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [tensorflow] | [build, test] | [ec2]

*Description:*
Fix the shm issue for TF HVD GPU  teset.

*Tests run:*
Local test passed.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/ec2.py [hunk 1]
```diff
@@ -361,9 +361,10 @@ def execute_ec2_training_test(connection, ecr_uri, test_cmd, region=DEFAULT_REGI
     connection.run(f"$(aws ecr get-login --no-include-email --region {region})", hide=True)
 
     # Run training command
+    shm_setting = '--shm-size="1g"' if large_shm else ""
```
Comment (TusharKanekiDey) [inline test/test_utils/ec2.py:364]: nit: should we keep it as gb instead of g ?

Comment (YYStreet) [reply test/test_utils/ec2.py:364]: Not sure if "gb" is an accepted and standard syntax.
Following the regulation on this page (one line ahead of this link line) https://docs.docker.com/engine/reference/run/#user-memory-constraints

File: test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py [hunk 2]
```diff
@@ -49,7 +49,8 @@ def test_tensorflow_train_mnist_cpu(tensorflow_training, ec2_connection, cpu_onl
 @pytest.mark.parametrize("ec2_instance_type", TF_EC2_GPU_INSTANCE_TYPE, indirect=True)
 def test_tensorflow_with_horovod_gpu(tensorflow_training, ec2_connection, gpu_only):
     test_script = TF1_HVD_CMD if is_tf1(tensorflow_training) else TF2_HVD_CMD
-    execute_ec2_training_test(ec2_connection, tensorflow_training, test_script)
+    execute_ec2_training_test(ec2_connection, tensorflow_training, test_script,
+                              large_shm=True if "p2.8xlarge" in TF_EC2_GPU_INSTANCE_TYPE else False)
```
Comment (nskool) [inline test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py:53]: Let's define a mapping between {'instance_type': 'shm_size'}, this mapping can reside in the ec2 utils. Passing large_shm=True will refer to the map.

**Comment by arjkesh** [comment]
nit: please update description and add comment in the ticket defining why we need to provision more shared memory

File: test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py (hunk 2 above)
Comment (YYStreet) [reply test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py:53]: > Let's define a mapping between {'instance_type': 'shm_size'}, this mapping can reside in the ec2 utils. Passing large_shm=True will refer to the map.

Sorry missed this comment. Already merged it in. Will address it in another PR

**Comment by YYStreet** [comment]
> nit: please update description and add comment in the ticket defining why we need to provision more shared memory

Will add more details in description. Thanks

**Pull request merged**